main.tf

```diff
@@ -208,7 +208,11 @@ data "google_storage_project_service_account" "gcs_account" {
 resource "google_project_iam_member" "eventarc_agent" {
   project = module.project_services.project_id
   member  = "serviceAccount:${google_project_service_identity.eventarc_agent.email}"
-  role    = "roles/eventarc.serviceAgent" # https://cloud.google.com/iam/docs/service-agents
+  for_each = toset([
+    "roles/eventarc.serviceAgent",             # https://cloud.google.com/iam/docs/service-agents
+    "roles/serviceusage.serviceUsageConsumer", # https://cloud.google.com/service-usage/docs/access-control
+  ])
+  role = each.key
```

Comment:
**Question:** Will this require updates to the list of roles specified in the .textproto (list of roles used by the service account created by Cloud Console)?
I don't think it will since this pull-request didn't require changes to https://github.com/GoogleCloudPlatform/terraform-genai-knowledge-base/blob/main/test/setup/iam.tf — but I want to double check with you.

Reply:
You're correct, only the ones to `test/setup/iam.tf` require changes to the Cloud Console service account (as far as I'm aware of)

Reply:
These roles are for the Eventarc agent, not for the solution service account.